tests/iconGenerator.test.js

```diff
@@ -20,6 +20,16 @@ describe('generating icons', function () {
     expect(iconGenerator.getPathToDirName(toDirName, fromDirPath)).toEqual(pathTo);
   });
 
+  it('file extension should not have a leading dot', function () {
+    files.supported
+      .filter(function (file) { return !file.filename; })
```

Comment:
I guess that you've chosen to favor readability instead of performance by double looping over the `files.supported` collection, haven`t you?